main.cpp

```diff
@@ -64,64 +64,95 @@ send me a DM to check your pull request
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
+    float value;
+    std::string name;
+    
+    T(float v, const char* nameptr)
+    {
+        value = v;
+        name = nameptr;
+    }   //1
```

Comment:
If you have to use a constructor that takes parameters, and those parameters initialize member variables, you should always prefer the member initialize list compared to initializing the variables inside the constructor body.

The reason is because the variables need to be instantiated/initialized when the class is created, and then when you assign to them in the ctor body, you are potentially re-initializing them.  that is a waste of resources.
- [ ] revise, use the member initializer list.